### PR TITLE
apache2.4 conf monitor-kraken

### PR DESCRIPTION
**Pull request by Hoshiyo**

### CODE REVIEW

File: fabfile/component/kraken.py [hunk 1]
```diff
@@ -60,8 +60,15 @@ def setup_kraken():
         use_sudo=True)
     update_monitor_configuration()
     if env.setup_apache:
-        _upload_template('kraken/monitor_apache_config.jinja', env.kraken_monitor_apache_config_file,
+        apache_conf_path = env.apache_conf_path('monitor-kraken')
+        _upload_template('kraken/monitor_apache_config.jinja', apache_conf_path,
                      context={'env': env}, backup=False)
+        if float(env.apache_version() >= 2.4) :
+            files.symlink(
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: I'm not an expert in apache, but isn't the 'right' way to call a2ensite ? maybe you can ask @francois-vincent  or @fdegomme-canaltp

File: fabfile/component/kraken.py [hunk 2]
```diff
@@ -60,8 +60,15 @@ def setup_kraken():
         use_sudo=True)
     update_monitor_configuration()
     if env.setup_apache:
-        _upload_template('kraken/monitor_apache_config.jinja', env.kraken_monitor_apache_config_file,
+        apache_conf_path = env.apache_conf_path('monitor-kraken')
+        _upload_template('kraken/monitor_apache_config.jinja', apache_conf_path,
                      context={'env': env}, backup=False)
+        if float(env.apache_version() >= 2.4) :
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: in term of programming I think its the responsibility of apache_version() to cast the result in float

File: fabfile/component/kraken.py (hunk 1 above)
Comment (Hoshiyo) [inline fabfile/component/kraken.py]: You right, and I knew it ...

File: templates/kraken/monitor_apache_config.jinja [hunk 3]
```diff
@@ -12,16 +12,15 @@ WSGIImportScript {{env.kraken_monitor_wsgi_file}} process-group=monitor-kraken a
 WSGIScriptAlias /{{env.kraken_monitor_location_dir}} {{env.kraken_monitor_wsgi_file}}
 SetEnv PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION cpp
 <Location /{{env.kraken_monitor_location_dir}}/>
+{% if env.apache_conf_path('monitor-kraken') == '/etc/apache2/conf-available/monitor-kraken.conf' %}
```
Comment (antoine-de) [inline templates/kraken/monitor_apache_config.jinja]: I don't understand this

File: fabfile/component/kraken.py (hunk 1 above)
Comment (fdegomme-canaltp) [inline fabfile/component/kraken.py]: I confirm : it is the right way :)

Comment (fdegomme-canaltp) [inline fabfile/component/kraken.py]: … BUT : a2ensite is just a perl script that make a symlink. So technically : it's the same thing.

**Comment by Hoshiyo** [comment]
Jenkins, test this please

File: templates/kraken/monitor_apache_config.jinja [hunk 4]
```diff
@@ -12,16 +12,15 @@ WSGIImportScript {{env.kraken_monitor_wsgi_file}} process-group=monitor-kraken a
 WSGIScriptAlias /{{env.kraken_monitor_location_dir}} {{env.kraken_monitor_wsgi_file}}
 SetEnv PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION cpp
 <Location /{{env.kraken_monitor_location_dir}}/>
+{% if  env.apache_version() >= 2.4 %}
```
Comment (antoine-de) [inline templates/kraken/monitor_apache_config.jinja]: isn't there a bad trailing slash ?

  <Location /{{env.kraken_monitor_location_dir}}/>

Comment (Hoshiyo) [inline templates/kraken/monitor_apache_config.jinja]: Where exactly? No, I don't think so.

**Comment by Hoshiyo** [comment]
@pbougue It's ready to be merged :innocent:

**Pull request merged**